modules/ROOT/pages/map-users-roles-ldap-task.adoc

```diff
@@ -1,39 +1,151 @@
+Anypoint Platform requires different information from your identity provider based on whether you use SAML, OpenID Connect, or LDAP.
 
-To configure a role:
+== Configure Multiple Group Mappings for Single Sign-On using SAML
+
+If your organization uses single sign-on and requires its users to obtain permissions from multiple groups, you can configure a group attribute that contains all the groups you want to use. In many cases, you can create an array of groups.
+
+For example, your IDP may provide your groups in the following format:
+
+----
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Org_Admin_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Exchange_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+----
+
+To prevent security vulnerabilities, Anypoint Platform requires you to create a single group that encompasses multiple group mappings:
+
+----
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Org_Admin_XXX</ns2:AttributeValue>
+  <ns2:AttributeValue>Mule_Exchange_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+----
+
+=== Map SAML Users to Roles
 
 . In Anypoint Platform, navigate to Access Management.
 . Click *Roles*.
 . Click *Add Role* to create a role for each group of users in your organization.
-+
-image::external-identity-34af9.png[]
-
 . Specify a role name and description.
-. Click *Add Role* to add the role:
-+
-image::external-identity-c731b.png[]
+. Click *Add Role* to add the role.
+. In the *Roles* section, click the name of the new role:
+. Click *Set external group mapping*.
+. Enter the your array of SAML groups.
+. Click *Set names*.
 +
+To map more than one attribute name to the selected role, click *Add More* and add another attribute.
+
+. Repeat this process for each role that you want to map to an external group.
+
+== Configure Multiple Group Mappings for Single Sign-On using OpenID Connect (OIDC)
+
+External identity providers format group mappings in different ways.
+Anypoint Platform checks the `UserInfo` endpoint for user information. If no user information is found, it then checks the `IdToken` endpoint.
 
+Okta, Auth0, Salesforce, Google, and Microsoft ADFS return groups that can be used for OIDC group mappings.
+You can use other identity providers if they call the `UserInfo` or `IdToken` endpoints.
+
+=== Map OIDC Group Users to Roles
+
+. Obtain a response to the `UserInfo` or `IdToken` endpoints with a group claim from your external identity provider.
+. In Anypoint Platform, navigate to Access Management.
+. Click *External Identity*.
+. From *Identity Management*, select OIDC.
+. Verify that the value in the *Group Scope* field matches the value in your IDP group expression.
```

Comment:
Group scope is a bit trickier in that it depends on the IdP and it's an optional field that is used in the request, it is not something that is visible in the response

```suggestion
. If your IdP requires an additional OIDC scope to authorize access to groups claims, it can be specified in the *Group Scope* field.
```